src/NexusMods.Games.FileHashes/FileHashesService.cs

```diff
@@ -0,0 +1,216 @@
+using System.IO.Compression;
+using System.Text.Json;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using NexusMods.Abstractions.Games.FileHashes;
+using NexusMods.Abstractions.Settings;
+using NexusMods.Games.FileHashes.DTOs;
+using NexusMods.MnemonicDB;
+using NexusMods.MnemonicDB.Abstractions;
+using NexusMods.MnemonicDB.Storage;
+using NexusMods.MnemonicDB.Storage.RocksDbBackend;
+using NexusMods.Paths;
+
+namespace NexusMods.Games.FileHashes;
+
+public class FileHashesService : IFileHashesService, IDisposable
+{
+    private readonly SemaphoreSlim _updateLock = new(1, 1);
+    private readonly FileHashesServiceSettings _settings;
+    private readonly IFileSystem _fileSystem;
+    private readonly HttpClient _httpClient;
+    private readonly JsonSerializerOptions _jsonSerializerOptions;
+    private readonly Dictionary<AbsolutePath, ConnectedDb> _databases;
+    private readonly IServiceProvider _provider;
+
+    /// <summary>
+    /// The currently connected database (if any)
+    /// </summary>
+    private ConnectedDb? _currentDb;
+
+    private readonly ILogger<FileHashesService> _logger;
+
+    private record ConnectedDb(IDb Db, Connection Connection, DatomStore Store, Backend Backend, DateTimeOffset Timestamp, AbsolutePath Path);
+
+    public FileHashesService(ILogger<FileHashesService> logger, ISettingsManager settingsManager, IFileSystem fileSystem, HttpClient httpClient, JsonSerializerOptions jsonSerializerOptions, IServiceProvider provider)
+    {
+        _logger = logger;
+        _httpClient = httpClient;
+        _jsonSerializerOptions = jsonSerializerOptions;
+        _fileSystem = fileSystem;
+        _settings = settingsManager.Get<FileHashesServiceSettings>();
+        _databases = new Dictionary<AbsolutePath, ConnectedDb>();
+        _provider = provider;
+        
+        _settings.HashDatabaseLocation.ToPath(_fileSystem).CreateDirectory();
+ 
+        // Delete any old databases that are not the latest
+        // we only delete at startup to avoid race conditions, but 
+        // the db doesn't update often so this is fine
+        foreach (var (_, path) in ExistingDBs().Skip(1))
+        {
+            path.DeleteDirectory(true);
+        }
+
+
+        
+        // Open the latest database
+        var latest = ExistingDBs().FirstOrDefault();
+        if (latest.Path != default(AbsolutePath))
+        {
+            _currentDb = OpenDb(latest.PublishTime, latest.Path);
+        }
+        
+        // Trigger an update
+        Task.Run(() => Task.FromResult(CheckForUpdate()));
```

Comment:
We have a `FireAndForget()` helper to catch exceptions in non awaited tasks like this.
Do you think it would be worth using here?

Reply:
I'll swap that over sometime soon